### DISCUSSION TITLE
International Pricing Issue

### DISCUSSION THREAD

**Discussion by AbhishekDubey987** [Shopify/storefront-api-feedback#97] [Help] [answered]
I am implementing international pricing in our Shopify app and it's working fine everywhere but when i am creating checkout i don't know why it is not giving me the converted price can anyone help me in this ?

I am using latest shopify sdk version
implementation ('com.shopify.mobilebuysdk:buy3:12.0.0'){
exclude group: "com.shopify.graphql.support"
}

here is the directive code which i am sending in createCheckout Function

  val directive: MutableList<Storefront.InContextDirective> = mutableListOf()
            var inContextDirective = Storefront.InContextDirective()
            inContextDirective.country= Storefront.CountryCode.IN
            directive.add(inContextDirective)

i am sending country code in mutation as you can see below

    fun createCheckout(
        inputs: Storefront.CheckoutCreateInput,
        directive: List<Storefront.InContextDirective>?=null
    ): Storefront.MutationQuery {
        var createcheckoutmutation: Storefront.MutationQuery? = null
        createcheckoutmutation = Storefront.mutation(directive) { root ->
            root
                .checkoutCreate(
                    inputs
                ) { checkoutquery ->
                    checkoutquery
                        .checkout { check ->
                            check
                                .webUrl()
                                .paymentDueV2({ pd -> pd.amount().currencyCode() })
                                .subtotalPriceV2({ st -> st.currencyCode().amount() })
                                .taxesIncluded()
                                .taxExempt()
                                .totalTaxV2({ tt -> tt.amount().currencyCode() })
                                .totalPriceV2({ tp -> tp.currencyCode().amount() })
                                .lineItems({ it.first(50) }, { args ->
                                    args.edges { egs ->
                                        egs.cursor()
                                            .node { linenode ->
                                                linenode
                                                    .title()
                                                    .quantity()
                                                    .variant({ linevariant ->
                                                        linevariant
                                                            .product({ p -> p })
                                                            .availableForSale()
                                                            .price()
                                                            .currentlyNotInStock()
                                                            .quantityAvailable()                           
                                                            .priceV2({ p ->
                                                                p.currencyCode().amount()
                                                                    .currencyCode()
                                                            })
                                                            .compareAtPriceV2({ c ->
                                                                c.amount().amount().currencyCode()
                                                            })
                                                            .image({ img ->
                                                                img
                                                                    .transformedSrc({ trans ->
                                                                        trans
                                                                            .maxHeight(200)
                                                                            .maxWidth(200)
                                                                    }
                                                                    )
                                                                    .originalSrc()
                                                            }
                                                            )
                                                            .selectedOptions({ select ->
                                                                select
                                                                    .name()
                                                                    .value()

                                                            }
                                                            )

                                                    }
                                                    )
                                            }
                                    }
                                })
                        }
                        .checkoutUserErrors { checkerror -> checkerror.field().message().code() }
                }
        }
        return createcheckoutmutation
    }

**Comment by johncraigcole** [comment, marked as answer]
Hey @AbhishekDubey987 👋 . To ensure the checkout object is in the currency and pricing you want you'll need to specify the buyer identity which is an input parameter to checkoutCreate. The `country` of the `inContext` directive is for read queries where you want to see pricing in a different currency like product queries.
